src/tera.rs

```diff
@@ -78,11 +78,26 @@ impl Tera {
             return Err(TeraError::InvalidValue(template_name.to_string()))
         }
 
-        let template = try!(self.get_template(template_name));
-        let mut renderer = Renderer::new(template, self, value);
+        let templates = try!(self.get_template_and_ancestors(template_name));
+        let mut renderer = Renderer::new(templates, self, value);
         renderer.render()
     }
 
+    fn get_template_and_ancestors(&self, template_name: &str) -> TeraResult<Vec<&Template>> {
```

Comment:
Can we build that list of dependencies at compile time? I.e. store ancestors in the `Template` struct.
We have `parent` already, we could fill `Template::ancestors` once all templates are parsed and that moves the `CircularExtends` error at compile time too.